exchanges/order/order_types.go

```diff
@@ -354,33 +348,42 @@ const (
 )
 
 // Type enforces a standard for order types across the code base
-type Type uint32
+type Type uint64
+
+// Is checks to see if the enum contains the flag
+func (t Type) Is(in Type) bool {
+	return in != 0 && t&in == in
+}
+
+// TimeInForceIs checks to see if the enum contains the time-in-force flag
+func (t Type) TimeInForceIs(in TimeInForce) bool {
+	return in != 0 && TimeInForce(t)&in == in
+}
 
 // Defined package order types
 const (
 	UnknownType Type = 0
 	Limit       Type = 1 << iota
 	Market
-	PostOnly
-	ImmediateOrCancel
 	Stop
-	StopLimit
-	StopMarket
 	TakeProfit
-	TakeProfitMarket
 	TrailingStop
-	FillOrKill
 	IOS
 	AnyType
 	Liquidation
 	Trigger
-	OptimalLimitIOC
-	OCO                              // One-cancels-the-other order
-	ConditionalStop                  // One-way stop order
-	MarketMakerProtection            // market-maker-protection used with portfolio margin mode. See https://www.okx.com/docs-v5/en/#order-book-trading-trade-post-place-order
-	MarketMakerProtectionAndPostOnly // market-maker-protection and post-only mode. Used in Okx exchange orders.
-	TWAP                             // time-weighted average price.
-	Chase                            // chase order. See https://www.okx.com/docs-v5/en/#order-book-trading-algo-trading-post-place-algo-order
+	OCO             // One-cancels-the-other order
+	ConditionalStop // One-way stop order
+	TWAP            // time-weighted average price.
+	Chase           // chase order. See https://www.okx.com/docs-v5/en/#order-book-trading-algo-trading-post-place-algo-order
+
+	// Hybrid order types
+	OptimalLimitIOC                  = 1<<iota | Type(ImmediateOrCancel)
+	MarketMakerProtection                                                     // market-maker-protection used with portfolio margin mode. See https://www.okx.com/docs-v5/en/#order-book-trading-trade-post-place-order
+	MarketMakerProtectionAndPostOnly = MarketMakerProtection | Type(PostOnly) // market-maker-protection and post-only mode. Used in Okx exchange orders.
+	StopLimit                        = Stop | Limit
+	StopMarket                       = Stop | Market
+	TakeProfitMarket                 = TakeProfit | Market
```

Comment:
I would suggest to keep TIF values as TIF and have it processed on the exchange side e.g.

OptimalLimitIOC -> OptimalLimit then exchange search for Is(order.ImmediateOrCancel)
MarketMakerProtectionAndPostOnly I would remove completely and do Is(order.PostOnly)

That way there is no chance of collision. The others can stay as is and it's less confusing.

Would be good to group these types in future to:

Type: Market, Limit
Conditional: Stop, TakeProfit, TrailingStop etc
ExchangeAlgo(or whatever name is best): TWAP, Chase, MarketMakerProtection
TimeInForce: What you have done so far.  

The above has some caveats but is more explicit and can be done step by step, the flagging and mixing could be a massive complex overhaul.